nutils/function.py

```diff
@@ -890,6 +890,7 @@ class _Gradient(Array):
 
   def __init__(self, func: Array, geom: Array) -> None:
     assert geom.spaces, '0d array'
+    assert geom.dtype != complex
```

Comment:
shouldn't we just assert that geom.dtype == float? (same in _SurfaceGradient, _Jacobian, _Normal, _ExteriorNormal)

Reply:
done